Lock down the trash collection more

johnswanson opened the pull request:
Oops. I was very thorough in checking that you couldn't modify the trash collection in the sense of moving things to or from it. I missed the most obvious thing: actually modifying the trash collection itself. So if someone wanted, they could change the name of the Trash to "Definitely Not the Trash".

Fixes #44817

johnswanson committed:
Lock down the trash collection more

Oops. I was very thorough in checking that you couldn't modify the
trash collection in the sense of moving things to or from it. I missed
the most obvious thing: actually modifying the trash collection itself.
So if someone wanted, they could change the name of the Trash to
"Definitely Not the Trash".

noahmoss reviewed (approved):
Stamping to unblock, but as a general question — should this be done at a lower level, like in the collection model's `define-before-update`?

johnswanson committed:
Move checks to the trash model itself

johnswanson committed:
Remove the Trash Collection from serialization

Serialization tests were failing due to the previous change, but this
got me thinking, and I realized that serializing the Trash collection is
completely unnecessary. We want to create it programatically on
startup (which we are) and then nothing should change it afterwards.

johnswanson commented:
@noahmoss I moved this to `define-before-update`, which caused test failures because when we deserialize the trash collection we end up calling update on it. This got me thinking about why we're serializing the trash collection in the first place if we never want it to change... so I removed serialization of it here. Seem reasonable to you?

noahmoss commented:
@johnswanson makes sense
